opendrift/readers/reader_netCDF_CF_unstructured.py

```diff
@@ -246,11 +246,16 @@ def get_variables(self,
         * salinity
 
         """
-        logger.debug("Requested variabels: %s, lengths: %d, %d, %d" %
-                     (requested_variables, len(x), len(y), len(z)))
 
         x = np.atleast_1d(x)
         y = np.atleast_1d(y)
+        z = np.atleast_1d(z)
```

Comment:
for `x` and `y` this is already done in variables.py:get_variables_interpolated_xy

Reply:
Ok, then we can also do the same for z there, and convert it to same length if scalar.

Reply:
Yes, reason I left it was in case logic in between required z = None for something. Think some readers choose depth themselves.

Reply:
This will be easier to decouple when we move `get_variables_impl` and `get_variables_derived` either above or below in the reader-stack.